### ISSUE TITLE
java.lang.ClassCastException: class org.javers.core.metamodel.property.MissingProperty cannot be cast to class java.util.List

### ISSUE THREAD

**Issue by tim-ber** [javers/javers#851]
Using javers-spring-boot-starter-sql 5.5.2, I'm getting the following error message for some types of commits. In the commit, a reference to another entity has been changed to null. Looks like `wrapNull` method is trying to cast an enum to a `List`, which throws `java.lang.ClassCastException`.

```
java.lang.ClassCastException: class org.javers.core.metamodel.property.MissingProperty cannot be cast to class java.util.List (org.javers.core.metamodel.property.MissingProperty is in unnamed module of loader 'app'; java.util.List is in module java.base of loader 'bootstrap')
	at org.javers.common.collections.Lists.wrapNull(Lists.java:20)
	at org.javers.core.metamodel.type.ListType.map(ListType.java:43)
	at org.javers.core.metamodel.type.EnumerableType.map(EnumerableType.java:37)
	at org.javers.core.graph.LiveNode.getDehydratedPropertyValue(LiveNode.java:85)
	at org.javers.core.diff.NodePair.getRightDehydratedPropertyValueAndSanitize(NodePair.java:40)
	at org.javers.core.diff.appenders.CorePropertyChangeAppender.calculateChanges(CorePropertyChangeAppender.java:44)
	at org.javers.core.diff.DiffFactory.appendChanges(DiffFactory.java:153)
	at org.javers.core.diff.DiffFactory.appendPropertyChanges(DiffFactory.java:143)
	at org.javers.core.diff.DiffFactory.createAndAppendChanges(DiffFactory.java:125)
	at org.javers.core.diff.DiffFactory.create(DiffFactory.java:69)
	at org.javers.core.commit.CommitFactory.createCommit(CommitFactory.java:89)
	at org.javers.core.commit.CommitFactory.create(CommitFactory.java:75)
	at org.javers.core.JaversCore.commit(JaversCore.java:82)
	at org.javers.spring.jpa.JaversTransactionalDecorator.commit(JaversTransactionalDecorator.java:82)
```

**Comment by bartoszwalacik** [comment]
Please write a failing test case, see how we write tests https://github.com/javers/javers/tree/master/javers-core/src/test/groovy/org/javers/core/cases

**Comment by bartoszwalacik** [comment]
looks like I have it

```groovy
    @TypeName("E")
    class Entity1 {
        @Id int id
    }

    @TypeName("E")
    class Entity2 {
        @Id int id
        List<String> props
    }

    def "should report when a list property is added or removed"(){
      given:
      def javers = javers().build()
      def object1 = new Entity1(id:1)
      def object2 = new Entity2(id:1, props: ["p"])

      when:
      def diff = javers.compare(object1, object2)
      println diff.prettyPrint()
      def changes = diff.getChangesByType(PropertyChange)

      then:
      changes.size() == 1
    }
```

**Comment by bartoszwalacik** [comment]
Fixed in 5.6.0
